lib/indexer/records.js

```diff
@@ -0,0 +1,152 @@
+/*!
+ * records.js - indexer records
+ * Copyright (c) 2018, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+/**
+ * @module lib/records
+ */
+
+const bio = require('bufio');
+const util = require('../utils/util');
+const consensus = require('../protocol/consensus');
+
+/**
+ * Block Meta
+ * @alias module:indexer.BlockMeta
+ */
+
+class BlockMeta {
+  /**
+   * Create block meta.
+   * @constructor
+   * @param {Hash} hash
+   * @param {Number} height
+   */
+
+  constructor(hash, height) {
+    this.hash = hash || consensus.NULL_HASH;
+    this.height = height != null ? height : -1;
+  }
+
+  /**
+   * Clone the block.
+   * @returns {BlockMeta}
+   */
+
+  clone() {
+    return new this.constructor(this.hash, this.height);
+  }
+
+  /**
+   * Get block meta hash as a buffer.
+   * @returns {Buffer}
+   */
+
+  toHash() {
+    return Buffer.from(this.hash, 'hex');
```

Comment:
If `this.hash` is already a  `Buffer`, there is no real need to pass it into `Buffer.from`

Reply:
Hmm.. maybe the intention was to copy the hash?